src/query/src/dist_plan/commutativity.rs

```diff
@@ -67,7 +82,17 @@ impl Categorizer {
             LogicalPlan::EmptyRelation(_) => Commutativity::NonCommutative,
             LogicalPlan::Subquery(_) => Commutativity::Unimplemented,
             LogicalPlan::SubqueryAlias(_) => Commutativity::Unimplemented,
-            LogicalPlan::Limit(_) => Commutativity::PartialCommutative,
+            LogicalPlan::Limit(limit) => {
+                // Only execute `fetch` on remote nodes.
+                // wait for https://github.com/apache/arrow-datafusion/pull/7669
+                if partition_cols.is_empty() && limit.fetch.is_some() {
```

Comment:
Is it possible for the `Limit` node to be commutative if we only fetch data from one datanode based on the query filter and partition rules?

Reply:
`partition_cols.is_empty()` is one of those cases. Infer from filter requires the filter to be strictly a superset of the partition rule, which is very limited and narrow. I don't give it a high priority to implement.

Reply:
>  Infer from filter requires the filter to be strictly a superset of the partition rule, which is very limited and narrow.

I don't think so, query data by partition columns is a common case. The user chooses the partition columns carefully.

Reply:
The partition values also have to match.

And this rule only cuts a few RPCs that won't return any rows. I don't think it's useless but there are many other rules that are much more useful.

Reply:
Ok, let's improve it step by step.